docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -252,6 +256,10 @@ Use the `setRowChildrenExpansion` method on `apiRef` to programmatically set the
 
 {{"demo": "RowGroupingSetChildrenExpansion.js", "bg": "inline", "defaultCodeOpen": false}}
 
+:::warning
+The `apiRef.current.setRowChildrenExpansion` method is not compatible with the [server-side tree data](/x/react-data-grid/server-side-data/tree-data/) and [server-side row grouping](/x/react-data-grid/server-side-data/row-grouping/). Use `apiRef.current.unstable_dataSource.fetchRows` instead.
```

Comment:
Curious, why it is not compatible? Could it call the data source to apply the behavior or is there a strong feasibility blocker?
(Question applies to both get and set methods)

Reply:
Data source changes a bit how expansion works.
When the user expands a row, `fetchRows` is called, which fetches the rows internally and only on success it calls `setRowChildrenExpansion` internally which only updates the `isExpanded` state of the respective `rowNode`.

If the user tries to call `setRowChildrenExpansion` there will be no rows to show, only the expand icon will shift to expanded form, thus `setRowChildrenExpansion` is no longer recommended to use with the data source.

Reply:
Good point @joserodolfofreitas 
@MBilalShafi Couldn't we potentially update `setRowChildrenExpansion` to fetch rows if the data source is being used?

Reply:
I missed the conversation here. 
Should we create an issue to make these approaches compatible?

Reply:
> Couldn't we potentially update setRowChildrenExpansion to fetch rows if the data source is being used?

We certainly could, but it hasn't been done in the current implementation because `setRowChildrenExpansion` sounds like (or anticipated as) a state setter (or a pure function), hiding the `fetch` part behind would be less aligned with a user's expectation as it could add fetch delays and request failures changing it's current behavior (impure).

Using it from within `fetchRows` makes sense if we understand it this way, or we may have to change its current perception.

What do you think?